File: android-start-activity/src/com/hmkcode/android/MainActivity.java
package com.hmkcode.android;

import android.os.Bundle;
import android.app.Activity;
import android.content.Intent;
import android.view.Menu;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;

public class MainActivity extends Activity implements OnClickListener {

	Button btnStartAnotherActivity;
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		
		btnStartAnotherActivity = (Button) findViewById(R.id.btnStartAnotherActivity);
		
		btnStartAnotherActivity.setOnClickListener(this);
	}


	@Override
	public void onClick(View view) {
		
		Intent inent = new Intent(this, AnotherActivity.class);

		// calling an activity using <intent-filter> action name 
		// 	Intent inent = new Intent("com.hmkcode.android.ANOTHER_ACTIVITY");
		
		startActivity(inent);
	}

}

File: intent-action-send/src/main/java/com/hmkcode/intent_action_send/MainActivity.kt
package com.hmkcode.intent_action_send

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private lateinit var textView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.button)
        textView = findViewById<TextView>(R.id.textView)

        button.setOnClickListener {
            textView.setText("Clicked!")
            send()
        }
    }

    fun send(){
        val sendIntent: Intent = Intent().apply {
            action = Intent.ACTION_SEND
            putExtra(Intent.EXTRA_TEXT, "Some text...")
            type = "text/plain"
        }

        val shareIntent = Intent.createChooser(sendIntent, null)
        startActivity(shareIntent)
    }
}
